src/providers/tpm/utils.rs

```diff
@@ -105,19 +105,23 @@ pub fn parsec_to_tpm_params(attributes: Attributes) -> Result<KeyParams> {
                 x @ 1024 | x @ 2048 | x @ 3072 | x @ 4096 => x.try_into().unwrap(), // will not fail on the matched values
                 _ => return Err(ResponseStatus::PsaErrorInvalidArgument),
             };
-            if attributes.is_encrypt_permitted() || attributes.is_decrypt_permitted() {
-                Ok(KeyParams::RsaEncrypt {
+            match attributes.policy.permitted_algorithms {
+                Algorithm::AsymmetricSignature(alg) if alg.is_rsa_alg() => Ok(KeyParams::RsaSign {
                     size,
+                    scheme: convert_asym_scheme_to_tpm(attributes.policy.permitted_algorithms)?,
                     pub_exponent: 0,
-                })
-            } else if attributes.is_hash_signable() || attributes.is_hash_verifiable() {
-                Ok(KeyParams::RsaSign {
+                }),
+                Algorithm::AsymmetricEncryption(_) => Ok(KeyParams::RsaEncrypt {
                     size,
-                    scheme: convert_asym_scheme_to_tpm(attributes.policy.permitted_algorithms)?,
                     pub_exponent: 0,
-                })
-            } else {
-                Err(ResponseStatus::PsaErrorNotSupported)
+                }),
+                alg => {
+                    error!(
+                        "Permitted algorithm {:?} not supported with RSA key pair.",
+                        alg
+                    );
+                    Err(ResponseStatus::PsaErrorNotSupported)
```

Comment:
I know this wasn't really supposed to be part of the PR anyway, but this should actually be `PsaErrorInvalidArgument` 😬 

> Implementations should return this error code when an enumeration parameter such as a key type, algorithm, etc. is not recognized. If a combination of parameters is recognized and identified as not valid, return PSA_ERROR_INVALID_ARGUMENT instead.